cmasher/utils.py

```diff
@@ -233,6 +239,97 @@ def _get_cmap_perceptual_rank(
 
 
 # %% FUNCTIONS
+# This function combines multiple colormaps at given nodes
+def combine_cmaps(
+    cmaps: list[Union[Colormap, str]],
+    nodes: list[float] = None,
+    n_rgb_levels: int = 256,
+) -> LinearSegmentedColormap:
+    """Create a composite matplotlib colormap by combining multiple colormaps.
+
+    Parameters:
+        cmaps (List[Union[Colormap, str]]): A list of colormaps to combine.
+        nodes (List[float], optional): The blending points between
+            colormaps, in the range [0, 1]. Defaults to equal divisions.
+        n_rgb_levels (int, optional): Number of RGB levels for each colormap
+            segment. Defaults to 256.
+
+    Returns:
+        Colormap: The composite colormap.
+
+    Raises:
+        TypeError: If the list contains mixed datatypes or invalid
+            colormap names.
+        ValueError: If the cmaps contain only one single colormap,
+            or if the number of nodes is not one less than the number
+            of colormaps, or if the nodes do not contain incrementing values
+            between 0.0 and 1.0.
+
+    Note:
+        The colormaps are combined from bottom to top.
+
+    References:
+        - https://stackoverflow.com/questions/31051488/combining-two-matplotlib-colormaps/31052741#31052741
+
+    Examples:
+        # Example 1: Using colormap names
+        >>> cmaps = ["ocean", "prism", "coolwarm"]
+        >>> nodes = [0.2, 0.75]
+        >>> custom_cmap_1 = combine_cmaps(cmaps, nodes)
+
+        # Example 2: Using predefined colormaps
+        >>> cmap_0 = plt.get_cmap("Blues")
+        >>> cmap_1 = plt.get_cmap("Oranges")
+        >>> cmap_2 = plt.get_cmap("Greens")
+        >>> custom_cmap_2 = combine_cmaps([cmap_0, cmap_1, cmap_2])
+    """
+
+    # Check colormap datatype and convert to List[Colormap]
+    if len(cmaps) <= 1:
+        raise ValueError("Cannot combine a single colormap.")
+    if all(isinstance(cmap, str) for cmap in cmaps):
+        cmaps = [mpl.colormaps[s] for s in cmaps]
```

Comment:
I would need your help to resolve this `mypy` error: `utils.py:291: error: Invalid index type "Colormap" for "ColormapRegistry"; expected type "str"  [index]` The type looks right to me, how could I get through this error?

Reply:
I get why you're confused, as the previous line correctly guards against incorrect usage. However, mypy isn't actually running code, and your guard condition is actually more complex than what `mypy` knows how to parse statically, so it just ignores it.
Actually, the logic is a bit broken too, as your annotation suggests that it should be possible to mix colormaps with colormap *names* within the `cmaps` argument. I'll add a suggestion to fix everything in one go.

Reply:
Thanks a lot for the suggested changes 👍. I have made those changes accordingly.

Reply:
I think there was a slight misunderstanding here: I meant to say that your annotation `List[str, Colormap]` didn't match the implementation, and wanted to suggest to change the implementation and keep the annotation. You actually changed both, so I think they still don't match 😅 (your new annotation matches your previous implementation and vice versa)

Reply:
Thanks for pointing out @neutrinoceros . Sorry this is a huge overlook from my side 😅. I will fix it now.